src/sisl/io/siesta/vibra.py

```diff
@@ -0,0 +1,257 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+from __future__ import annotations
+
+import numpy as np
+from itertools import product
+
+from sisl import Geometry, Lattice, constant
+from sisl._internal import set_module
+from sisl.messages import warn
+from sisl.physics.phonon import EigenmodePhonon
+from sisl.unit.siesta import unit_convert
+
+from ..sile import add_sile, sile_fh_open, sile_raise_write
+from .sile import SileSiesta
+
+
+__all__ = ["vectorsSileSiesta"]
+
+_Bohr2Ang = unit_convert("Bohr", "Ang")
+_J2eV = unit_convert("J", "eV")
+_hc_eV = constant.h * constant.c * _J2eV
+
+@set_module("sisl.io.siesta")
+class vectorsSileSiesta(SileSiesta):
+    """Phonon eigenmode file"""
```

Comment:
Argument details so it is clear that `geometry`/`parent=Hessian` would be required.